<p>Fields which are known to contain personally identifiable financial information (PIFI) should not be written to logs. The data must be properly encrypted at all times for our customer's safety and to ensure we're properly compliant with regulations.  Any PIFI fields which are found in log files will be subject to additional auding which is time consuming and costly.</p>

<h2>Noncompliant Code Example</h2>
<pre>
  log-manager:write-message("Debug1").
  find first Employee no-lock where Employee.last-name = "Herring" no-error.
  if available Employee then
    //Issue - field has "[PIFI]" in the description
    log-manager:write=message("Debug2: " + Employee.EmpNum + " " + Employyee.FirstName).
  else
    log-manager:write-message("Debug3: "No employee found...").
</pre>

<h2>Compliant Solution</h2>
<pre>
    log-manager:write-message("Debug1").
    find first Employee no-lock where Employee.last-name = "Herring" no-error.
    if available Employee then
      //Issue - field has "[PIFI]" in the description"
      log-manager:write=message("Debug2: Found at least one employee record").
    else
      log-manager:write-message("Debug3: "No employee found...").
</pre>
